shell.nix

```diff
@@ -0,0 +1,24 @@
+{ args ? {} , shell ? "bash" }:
+let
+  shared = import ./nix/shared.nix args;
+  static = shared.possibly-static;
+in
+shared.nixpkgs.runCommand "dhall-shell-bash" {
+  buildInputs = [
+    static.dhall
+    static.dhall-json
+    static.dhall-text
+    static.dhall-bash
+    static.dhall-nix
+  ];
+  shellHook = ''
+    echo -e "\e[0;32m""\nA Dhall shell""\e[0m"
```

Comment:
The user probably already knows that this is a Dhall shell of some form so this message does not add new information.  Perhaps remove it or replace with a message providing instructions for how to use the shell

Reply:
Initially just deleted it but then decided to add some argument examples. Not quite sure if it is necessary. (I do normally keep such examples even for own usage in the comments in `source.nix`). Pls let me know what you think.

Reply:
Yeah, this looks good

Reply:
Added a short paragraph about the top-level `nix-shell` usage in README.md if it is ok.

Reply:
Yeah, good idea!